### PR TITLE
Fix #372: add tests to verify $date-value updates work as-is

### LINKED ISSUE #372: (Date/Time support) Allow update operators on Date/Time valued fields
As part of Date/Time (#362) support we need to ensure that existing update operators applicable to this type (generally same as other non-numeric types) can be used with Date/Time fields.
This may require changes to update operators, especially validation (since pseudo-field name `$date` needs to be allowed and not taken as update operator); but even without other changes need to add testing.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds tests (unit, integration) to verify that Date-valued fields can be updated.

**Which issue(s) this PR fixes**:
Fixes #372

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/UpdateClauseDeserializerTest.java [hunk 1]
```diff
@@ -123,5 +123,26 @@ public void mustHandleSubdoc() throws Exception {
       UpdateClause updateClause = objectMapper.readValue(json, UpdateClause.class);
       assertThat(updateClause.buildOperations()).hasSize(1).contains(operation);
     }
+
+    @Test
+    public void mustHandleDate() throws Exception {
```
Comment (maheshrajamani) [inline src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/UpdateClauseDeserializerTest.java:128]: Can you add a  test case to show invalid date format is handled.
`{"$set" : {"dateType": {"$date": "2023-01-01T00:00:00Z"}}}`

**Review by maheshrajamani — APPROVED** [review]
LGTM

File: src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/UpdateClauseDeserializerTest.java (hunk 1 above)
Comment (tatu-at-datastax) [reply src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/UpdateClauseDeserializerTest.java:128]: I can do that. It needs to go in integration test I think, but I'll see.

Comment (tatu-at-datastax) [reply src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/UpdateClauseDeserializerTest.java:128]: Added an IT (validation occurs in Shredder, not yet during deserialization)

**Pull request merged**